src/pykeen/nn/init.py

```diff
@@ -311,6 +312,68 @@ def from_triples_factory(
         )
 
 
+class WeisfeilerLehmanInitializer:
+    """An initializer based on an encoding of categorical colors from the Weisfeiler-Lehman algorithm."""
+
+    def __init__(
+        self,
+        *,
+        # the color initializer
+        color_initializer: Hint[Initializer] = None,
+        color_initializer_kwargs: OptionalKwargs = None,
+        # variants for the edge index
+        edge_index: Optional[torch.LongTensor] = None,
+        num_entities: Optional[int] = None,
+        mapped_triples: Optional[torch.LongTensor] = None,
+        triples_factory: Optional[CoreTriplesFactory] = None,
+        # additional parameters for iter_weisfeiler_lehman
+        **kwargs,
+    ) -> None:
+        """
+        Initialize the initializer.
+
+        :param color_initializer:
+            the initializer for initialization color representations, or a hint thereof
+        :param color_initializer_kwargs:
+            additional keyword-based parameters for the color initializer
+
+        :param edge_index: shape: (2, m)
+            the edge index
+        :param num_entities:
+            the number of entities. can be inferred
+        :param mapped_triples: shape: (m, 3)
+            the Id-based triples
+        :param triples_factory:
+            the triples factory
+
+        :param kwargs:
+            additional keyword-based parameters passed to :func:`pykeen.utils.iter_weisfeiler_lehman`
+        """
+        edge_index = get_edge_index(
```

Comment:
at second thought, I'm a bit hesitant about this API. maybe we should make the `__init__` only take in the edge index and have alternate class method constructors for triples factories and mapped triples

Reply:
separate factory methods are not easy to use with `initializer_kwargs`, though.

Reply:
okay then leave it